src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,274 @@
+@namespace("org.ga4gh.methods")
+protocol AlleleAnnotationMethods {
+
+/*This protocol defines methods use to mine pre-calculated annotation sets.*/
+
+import idl "methods.avdl";
+import idl "variants.avdl";
+import idl "alleleAnnotations.avdl";
+import idl "sequenceAnnotations.avdl";
+import idl "ontologies.avdl";
+
+/*
+variantannotations/search returns annotation for the alleles of classic/linear variants
+alleleannotations/search returns annotation for graph alleles
+
+Both allow the mining of allele-specific annotations by either a region or a set of genomic
+features. Where a region is supplied annotation of all alleles vs all features in the region
+is returned. Where a set of features is supplied, only annotations against these features
+(matching on featuretype and id) are returned and other overlapping features are ignored.
+
+annotationsets/search returns information on reference data and software versions used in
+calculating the annotation. It is essential this information is exhaustive.
+*/
+
+/******************  /variantannotations/search  *********************/
+/** This request maps to the body of `POST /variantannotations/search` as JSON. */
+record SearchVariantAnnotationsRequest {
+  /** Required. The ID of the annotation set to search over. */
+  string annotationSetId;
+
+  /**
+  Only return annotation for variants which have exactly this name
+  (case-sensitive, exact match).
+  */
+  union { null, string } name = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this name.
+  One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this ID. One
+  of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The beginning of the window (0-based, inclusive) for which variants with
+  overlapping reference alleles should be returned.
+  Genomic positions are non-negative integers less than reference length.
+  Requests spanning the join of circular genomes are represented as
+  two requests one on each side of the join (position 0).
+  */
+  long start;
+
+  /**
+  Required if referenceName or referenceId supplied.
+  The end of the window (0-based, exclusive) for which variants with
+  overlapping reference alleles should be returned.
+  */
+  long end;
+
+
+  /**
+  Only return variant annotations for any of these features
+  Features may include specific transcripts, genes or regulatory elements
+  This or a location (referenceName/referenceId plus start and end) must be supplied
+  If null, return all variant annotations in specified window.
+  */
+  union { null, array<org.ga4gh.models.Feature> } features;
```

Comment:
yes, reference by ids rather than copying features will results
a simpler data model.  

CH Albach <notifications@github.com> writes:
> I still think it's too confusing to embed the entire feature here. The
> coordinates here will have a very complicated interaction with the above
> proposed referenceName, start, end.
> 
> feature_ids I think would be better, or just remove this field entirely for the
> first draft.
> 
> —
> Reply to this email directly or view it on GitHub.*
>